utils/general.py

```diff
@@ -36,6 +38,69 @@
 # Prevent OpenCV from multithreading (to use PyTorch DataLoader)
 cv2.setNumThreads(0)
 
+# Download objects
+def download_file_blob_to_dir(directory, file_blob):
+    file_name = file_blob.name.split("/")[2]
+    destination = Path(directory) / f"{file_name}"
+    destination.touch()
+    blob.download_to_filename(destination)
+
+
+# Place GCS objects in their respective local directories
+def place_object(local_directory, file_blob):
+
+    train_image_directory = local_directory[0]
+    train_label_directory = local_directory[1]
+    validation_image_directory = local_directory[2]
+    validation_label_directory = local_directory[3]
+
+    if file_blob.name.startswith("images/t") and file_blob.name.endswith(".jpg"):
+        download_file_blob_to_dir(train_image_directory, file_blobblob)
+
+    elif file_blob.name.startswith("labels/t") and file_blob.name.endswith(".txt"):
+        download_file_blob_to_dir(train_label_directory, file_blob)
+
+    elif file_blob.name.startswith("images/v") and file_blob.name.endswith(".jpg"):
+        download_file_blob_to_dir(validation_image_directory, file_blob)
+
+    elif file_blob.name.startswith("labels/v") and file_blob.name.endswith(".txt"):
+        download_file_blob_to_dir(validation_label_directory, file_blob)
+
+
+# Decorator for "check_dataset"
+def download_and_setup_dataset(func):
```

Comment:
I would probably be a little more explicit. The decorator downloads data from Google Cloud, so the name should reflect that.

Reply:
Fixed 9100222e106d4493d1cdb74a9063b1445d5aa733

Reply:
Fixed 9100222e106d4493d1cdb74a9063b1445d5aa733